Feature/#110 사이드바 반응형 수정

llddang opened the pull request:
### 관련 이슈

- close #110

### 작업 요약

- 사이드바에 들어가는 요소들을 반응형으로 수정

### 리뷰 요구 사항
- 각자의 화면에서 사이드바의 크기가 이상한지 한 번 확인 부탁드립니다!
- 제 노트북에서 크기를 줄여가며 확인해보았는데요, 정확한 width*height 비율이 아니라서 다른 분들 노트북에서는 어떻게 되는지 궁금합니다.
- 리뷰예상 시간 : `3분`

### 미리 보기

- 영상을 올리려고 했는데요, 10MB가 넘어가서 못 올립니당!
- 각자 노트북에서 확인부탁드려요!

llddang committed:
design: 반응형을 위한 font style 적용

#110

llddang committed:
feat: 사이드바의 CaregoryButton의 isTeam, isTeamMatch props의 기본값을 false로 주기.

#110

llddang committed:
chore: 반응형에 따라서 크기가 변경되도록 icon button size 재설정

#112

llddang committed:
Merge branch 'feature/#112-반응형_버튼_크기_지정' of https://github.com/BDD-CLUB/01-doo-re-front into feature/#110-사이드바_반응형_수정

llddang committed:
refactor: 사이드바의 햄버거 크기 설정을 좀 더 보기 좋게 변경함.

#110

llddang committed:
design: side bar에서 width 속성 삭제

- width이 고정이 아닌, fit-content로 사용됨.
#110

llddang committed:
Merge branch 'develop' of https://github.com/BDD-CLUB/01-doo-re-front into feature/#110-사이드바_반응형_수정

# Conflicts:
#	src/theme/components/button.ts

pipisebastian commented on src/components/Sidebar/index.tsx:
```diff
@@ -16,26 +15,24 @@ const Sidebar = () => {
-    <Flex
-      direction="column"
-      gap="16"
-      w={isOpen ? size.sidebarWidth : '16'}
```
sidebarWidth안 쓰고 아예 내부 요소 넓이에 맞춰서 사이드바 넓이가 줄어드는군요! 좋은 것 같습니다! 혹시 그렇다면 constant/size.ts에 있는 sidebarWidth는 아예 지워주실 수 있을까용?

pipisebastian reviewed (approved):
수고하셨습니다! 그런데 애니메이션이 안먹는 것 같습니당.. width를 기준으로 주는데, width가 명시적으로 안적혀있어서 그런게 아닐까 하네용...

mun-kyeong commented on src/components/Sidebar/Button/CategoryButton/index.tsx:
```diff
@@ -3,11 +3,12 @@ import { BsCircleFill } from 'react-icons/bs';
 import { CategoryButtonProps } from '../../type';
 
-const CategoryButton = ({ path, text, isSelected, isTeam, isTeamMatch }: CategoryButtonProps) => {
+const CategoryButton = ({ path, text, isSelected, isTeam = false, isTeamMatch = false }: CategoryButtonProps) => {
```
옹 defalut값 넣는거 좋습니다!!

mun-kyeong commented on src/components/Sidebar/index.tsx:
```diff
@@ -16,26 +15,24 @@ const Sidebar = () => {
-      bg="green"
-      transition="width 0.15s ease-in-out"
-    >
+    <Flex direction="column" gap="16" px={isOpen ? '4' : '2'} py="4" bg="green" transition="width 0.15s ease-in-out">
```
<img width="959" alt="image" src="https://github.com/BDD-CLUB/01-doo-re-front/assets/96461334/f0dcd49e-a97a-4db0-b751-9f471b54aeb0">

이거 이 현상이 일어나는 이유가 gap="16" 때문이더군요! 
제 노트북 화면이 작은 편이라 gap이 너무 커서 화면 높이가 강제로 조절되는 것 같습니다

```javascript
      gap={{ xl: '4', '2xl': '16' }}
```

이코드로 바꾸니까

<img width="960" alt="image" src="https://github.com/BDD-CLUB/01-doo-re-front/assets/96461334/5952dee2-e3c0-4ec1-be37-b9fa49c6bcab">

이렇게 강제로 조절되는 높이가 줄어드는걸 확인할 수 있습니당
위의 사진은 제 노트북 화면 기준으로 100% 화면 적용해서 사진 보여드린거구요,
현재 해상도는 1920 x 1080입니닷

제 화면이 작은편이라 굳이 바꿔야 하는 부분은 아닐수도 있겠지만, 
그래도 한번 검토해주십사- 하고 코멘트 남깁니당 :)

mun-kyeong commented on src/components/Sidebar/index.tsx:
```diff
@@ -59,8 +56,8 @@ const Sidebar = () => {
-          <Flex align="center" justify="space-between">
-            <Text color="white" fontSize="xl" fontWeight="bold">
+          <Flex align="center" justify="space-between" gap="2">
+            <Text textStyle="bold_xl" color="white" whiteSpace="nowrap">
```
옹 figma에 있는text 속성을 이제 그대로 쓸 수 있군요 
이전에는 pr 확인이 늦어서 바뀐 부분을 몰랐는데 이제부턴 이런 방식으로 써야겠어욤 :)

pipisebastian replied:
네넹! 이부분은 슬랙에 공지올리는걸 까먹었군요.. 올리겠습니다!

mun-kyeong reviewed (commented):
좋습니다!! 수고하셨어요
코멘트 남긴부분 확인부탁드립니닷

+) 피피님이 말씀하신 것과 같이 애니매이션 적용 안되는 것 같습니다..!

pipisebastian committed:
design: 사이드바 모바일 반응형 구현

#110

pipisebastian commented on src/components/Sidebar/index.tsx:
```diff
@@ -1,76 +1,28 @@
+    if (!isDesktop) {
+      setIsOpen(false);
+    }
+  }, [isDesktop]);
```
이렇게 한 이유는, 사이드바를 펼친 상태 + 모바일(대충 768px)일때는 회색 배경화면이 같이 보이는데, 조금 어색해보여서
일부로 모바일일때 사이드바를 닫히게 했습니다!

pipisebastian commented on src/components/Sidebar/index.tsx:
```diff
@@ -1,76 +1,28 @@
-        </Card>
-      )}
-    </Flex>
+  return isDesktop || !isOpen ? (
```
그리고 조건문이 이런 이유는
닫혀있을때는 무조건 옆으로 밀려야합니다. 즉 닫힌 사이드바가 컨텐츠 위에 absolute되면, 컨텐츠가 가려져서 어색해집니다,
즉, 열렸을때 + 모바일일때만, absolute처리 해줘야해서 이렇게 했습니다!

yeonddori reviewed (commented):
![스크린샷 2024-03-03 224045](https://github.com/BDD-CLUB/01-doo-re-front/assets/126975394/6f21213c-c7d4-4d24-9180-8b0445c3ae56)
![스크린샷 2024-03-03 224052](https://github.com/BDD-CLUB/01-doo-re-front/assets/126975394/7f8e9bd3-35f7-4644-8bd1-93463b22f12f)

모바일에서 열릴 때만 absolute 한다고 하셨는데, 사진에서 말풍선을 보시면 좀 더 직관적이게 알 수 있듯이 열릴 때 페이지의 내용 위에 자연스럽게 겹치는 게 아니라 접혔을 때의 사이드바 길이만큼 페이지 내용이 왼쪽으로 밀리면서 열리게 되는데 혹시 이건 의도된 바인지 궁금합니다!

\+ 이제 보니 이 문제는 아래 문제로 인한 현상 같기도 하네요!

yeonddori commented on src/components/Sidebar/index.tsx:
```diff
@@ -1,76 +1,28 @@
-        <Flex direction={isOpen ? 'row' : 'column'} gap="4">
-          <SidebarIconButton icon={<BiBell />} onClick={() => {}} />
-          <SidebarIconButton icon={<BiUser />} onClick={() => {}} />
-          <SidebarIconButton icon={<MdOutlineLogout />} onClick={() => {}} />
-        </Flex>
-      </Flex>
+  useEffect(() => {
+    if (!isDesktop) {
+      setIsOpen(false);
+    }
+  }, [isDesktop]);
```
이 부분이 제대로 동작을 하지 않습니다!
` const isDesktop = useMediaQuery('(min-width: 768px)');`로 isDesktop에 괄호를 빼면 데스크탑일 때 제대로 작동하지만 또 모바일일 때는 제대로 작동하지가 않네요.. 아직 원인을 제대로 모르겠습니다 ㅜ

pipisebastian replied:
앗 혹시 어떤 동작말하는걸까요?
의도한거는 desktop에서 사이드바를 펼친 상태에서, 모바일이 되었을때 자동으로 사이드바가 닫히는 건데, 이게 안된다는 걸까요? ㅜㅜ

yeonddori replied:
제가 이해한 바로는 데스크탑일 때 초기 상태는 사이드바 열림 상태이고, 모바일일 때는 피피님 말씀처럼 사이드바를 닫게 하는 것이 제대로 된 동작이라고 이해하였습니다! isOpen의 초기값이 true이니까요! 근데 제 화면에서는 데스크탑에서도, 모바일에서도 항상 닫힘 상태가 유지가 됩니다.. 그래서 저는 제대로 동작하지 않는다고 생각했어요! 혹시 제가 제대로 이해를 못한 것이었을까요..🥲

pipisebastian replied:
아하 맞네요,,! 초기상태 열린거는 생각 못했습니다! 감사합니다!

찾아봤는데, 맨 처음에는 isDesktop가 false였다가, 시간이 좀 지난뒤에 true로 되더라구요!
그래서 isDesktop가 false인 시간동안, useEffect가 돌아서 isOpen을 false로 만드는 바람에, 초기에도 사이드바가 닫힌 상태였던 것 같습니다!
해결은 [차크라 ui ](https://chakra-ui.com/docs/hooks/use-media-query) 에서 isDisplayingInBrowser 상태값을 하나 더 가져와서, useEffect가 도는 조건에 isDisplayingInBrowser가 true인지도 같이 판별해주면 되긴한데, 

음.. 이렇게 되더라도 발견한 문제가 하나 더 있는데,, 이게 isDesktop이 어쨌든 처음에는 false여서, 렌더링 초기에 회색배경이 깔린게 맨 처음에 로딩되고, 그다음에 정상적으로 렌더링되더라구요! 즉 회색배경이 있다가 없어지는게 눈에 보여서, 어색하네요..

이거  좀 더 찾아보고 다시 리뷰요청 드려도 될까요?

yeonddori replied:
물론이죠! 생각보다 더 복잡한 일이었네요 ㅜ 저도 시간 날 때 해결 방법 같이 찾아볼게요~!
저는 30분 고민하는 동안 원인조차 몰랐었는데, 피피님은 바로 찾으시다니 대단하세요👍

pipisebastian commented:
> ![스크린샷 2024-03-03 224045](https://private-user-images.githubusercontent.com/126975394/309554142-6f21213c-c7d4-4d24-9180-8b0445c3ae56.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk0ODA3MTMsIm5iZiI6MTcwOTQ4MDQxMywicGF0aCI6Ii8xMjY5NzUzOTQvMzA5NTU0MTQyLTZmMjEyMTNjLWM3ZDQtNGQyNC05MTgwLThiMDQ0NWMzYWU1Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwM1QxNTQwMTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mNGI4YzZiNmExMzJmY2M5ZjBiNjA1OTJkNjE2YmVjYTkzOTM5MWE3MmUxZGU5M2VjYjI4OTBkYzgzNWU5NzgyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.DLSEAwmGa8F2k0ZvNVFoor5sx7c2Rq5SJ7PZl0RckZU) ![스크린샷 2024-03-03 224052](https://private-user-images.githubusercontent.com/126975394/309554145-7f8e9bd3-35f7-4644-8bd1-93463b22f12f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk0ODA3MTMsIm5iZiI6MTcwOTQ4MDQxMywicGF0aCI6Ii8xMjY5NzUzOTQvMzA5NTU0MTQ1LTdmOGU5YmQzLTM1ZjctNDY0NC04YmQxLTkzNDYzYjIyZjEyZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwM1QxNTQwMTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kNjA3NzUzOGQxOGY5NTk2ZWU2YTI5YmFlOGViOGQ5YjY3YWFhNjhlY2IwYTZiNzI4ODQzMjhlMTI1OTRlZTY0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.j4z3oPf0WDi1gzjC07daYaZkfIIdCghlyPuNuEo80hM)
> 
> 모바일에서 열릴 때만 absolute 한다고 하셨는데, 사진에서 말풍선을 보시면 좀 더 직관적이게 알 수 있듯이 열릴 때 페이지의 내용 위에 자연스럽게 겹치는 게 아니라 접혔을 때의 사이드바 길이만큼 페이지 내용이 왼쪽으로 밀리면서 열리게 되는데 혹시 이건 의도된 바인지 궁금합니다!
> 
> + 이제 보니 이 문제는 아래 문제로 인한 현상 같기도 하네요!

앗 들켰군요.. 사실 저 문제가 모바일+사이드바 닫혔을때는, 컨텐츠 부분이 사이드바 넓이만큼 밀려야하는데,
모바일+사이드바 열렸을때도 (닫힌 사이드바 넓이만큼 밀려야 하는데) 안밀려서 그렇습니다! 도리님이 말씀하신 그대로에요!

일단 닫힌 사이드바 넓이, 50px만큼 옆으로 밀려주도록 코드를 수정했는데, 혹시 더 좋은 방법있으시면 추천부탁드립니다! 코드가 맘에 들지 않습니다..

pipisebastian committed:
design: 모바일 사이드바 열려있을때 내부 콘텐츠 밀리도록 수정
#110

llddang committed:
feat: 화면의 크기에 따른 반응형 조절 방식 변경

- 110

pipisebastian committed:
design: 사이드바 초기상태 닫히도록 수정 및 배경색 추가
#110

llddang committed:
fix: 사이드바 zindex 범위 조정

- 사이드 바 전체에 zIndex = 99로 설정하니, 뒤에 있는 요소들을 선택할 수가 없음.
- 그래서 화면의 width이 768px보다 작고 사이드바가 펼쳐졌을 때만, zIndex = 2로 설정.
- #110

llddang committed:
refactor: 사이드바 배경과, 사이드바 통합.

- #110

jasper200207 reviewed (approved):
👍
